app/models/adventure.rb

```diff
@@ -20,6 +22,12 @@ class Adventure < ActiveRecord::Base
   validates :latitude, numericality: { only_float: true, allow_blank: true }
   validates :longitude, numericality: { only_float: true, allow_blank: true }
 
+  geocoded_by :address
+  after_validation :geocode  # auto-fetch coordinates
+  reverse_geocoded_by :latitude, :longitude, address: :address
+  after_validation :reverse_geocode  # auto-fetch address
+  after_validation :geocode, if: ->(obj){ obj.address.present? and obj.address_changed? }
```

Comment:
Line is too long. [89/80]<br>Use `&&` instead of `and`.<br>Space missing to the left of {.